pylint/config/option_manager_mixin.py

```diff
@@ -292,36 +300,53 @@ def read_config_file(self, config_file=None, verbose=None):
             msg = "No config file found, using default configuration"
         print(msg, file=sys.stderr)
 
-    @staticmethod
     def _parse_toml(
-        config_file: Union[Path, str], parser: configparser.ConfigParser
+        self, config_file: Union[Path, str], parser: configparser.ConfigParser
     ) -> None:
         """Parse and handle errors of a toml configuration file."""
         with open(config_file, encoding="utf-8") as fp:
             content = toml.load(fp)
         try:
             sections_values = content["tool"]["pylint"]
         except KeyError:
-            pass
-        else:
-            for section, values in sections_values.items():
-                # TOML has rich types, convert values to
-                # strings as ConfigParser expects.
-                for option, value in values.items():
-                    if isinstance(value, bool):
-                        values[option] = "yes" if value else "no"
-                    elif isinstance(value, (int, float)):
-                        values[option] = str(value)
-                    elif isinstance(value, list):
-                        values[option] = ",".join(value)
-                parser._sections[section.upper()] = values  # type: ignore
+            return
+        for section, values in sections_values.items():
+            section_name = section.upper()
+            # TOML has rich types, convert values to
+            # strings as ConfigParser expects.
+            if not isinstance(values, dict):
+                # This class is a mixin the add_message come from the pylinter
+                self.add_message(  # type: ignore
+                    "bad-configuration-option-value", line=0, args=(section, values)
+                )
+                continue
+            for option, value in values.items():
+                if isinstance(value, bool):
+                    values[option] = "yes" if value else "no"
+                elif isinstance(value, list):
+                    values[option] = ",".join(value)
+                else:
+                    values[option] = str(value)
+            for option, value in values.items():
+                try:
+                    parser.set(section_name, option, value=value)
+                except configparser.NoSectionError:
+                    parser.add_section(section_name)
+                    parser.set(section_name, option, value=value)
 
     def load_config_file(self):
         """Dispatch values previously read from a configuration file to each
         options provider)"""
         parser = self.cfgfile_parser
         for section in parser.sections():
-            for option, value in parser.items(section):
+            try:
+                items = parser.items(section)
+            except ValueError:
+                self.add_message(
```

Comment:
Might want to add an explanation for why we get here.

Can we get the incorrect item from the ValueError message? If so we might be able to pass it as an `args`.